docs/guides/web-api-basics.md

```diff
@@ -0,0 +1,444 @@
+
+```java
+import com.slack.api.model.Message;
+
+ChatPostMessageResponse response = slack.methods(token).chatPostMessage(req -> req
+  .channel("C1234567")
+  .text("Write one, post anywhere"));
+if (response.isOk()) {
+  Message postedMessage = response.getMessage();
+} else {
+  String errorCode = response.getError(); // e.g., "invalid_auth", "channel_not_found"
+}
+```
+
+Not only the case you receive a successful response containing `"ok": false` property at the top level, errors can happen for many reasons. You need to be aware of the following three error patterns when using this SDK.
+
+1. Received a successful response but in its body, `"ok"` is **false** and an `"error"` such as `channel_not_found` exists
+2. Got a **java.io.IOException** thrown due to connectivity issues
+3. Got a **com.slack.api.methods.SlackApiException** thrown for an unsuccessful response
+
+To understand how to handle **1.** pattern, read [this API document](https://api.slack.com/web#evaluating_responses).
+
+As for **2.** & **3.** patterns, the **MethodsClient** may throw two types of exceptions. Applications using this library is responsible for catching and handling these exceptions properly.
+
+|Exception|Information Included|Reason|
+|-|-|-|
+|**java.io.IOException**|Has only standard exception information - string message and cause|This exception can be thrown if the request could not be executed due to cancellation, a connectivity problem or timeout.|
+|**com.slack.api.methods.SlackApiException**|Has underlying HTTP response, raw string response body, and deserialized **SlackApiErrorResponse** object|This exception can be thrown if Slack API servers respond with an unsuccessful HTTP status code (not 20x).|
+
+The final form of well-considered error handling would be the one below.
+
+```java
+import java.io.IOException;
+import com.slack.api.methods.SlackApiException;
+
+try {
+  ChatPostMessageResponse response = slack.methods(token).chatPostMessage(req -> req
+    .channel("C1234567")
+    .text("Write one, post anywhere"));
+  if (response.isOk()) {
+    Message postedMessage = response.getMessage();
+  } else {
+    String errorCode = response.getError(); // e.g., "invalid_auth", "channel_not_found"
+  }
+} catch (SlackApiException requestFailure) {
+  // Slack API responded with unsuccessful status code (= not 20x)  
+} catch (IOException connectivityIssue) {
+  // Throwing this exception indicates your app or Slack servers had a connectivity issue.
+}
+```
+
+## There's More!
+
+Slack Web API offers [180+ methods](https://api.slack.com/methods). The way to use others is almost the same. Just calling methods in **MethodsClient** with a valid token and sufficient parameters works for you.
+
+A good way to check the entire list of methods available in this SDK is to access [the Javadoc](https://javadoc.io/doc/com.slack/slack-api-client/latest/com/slack/api/methods/MethodsClient.html).
```

Comment:
This link doesn't work - is it going to go live when docs are distributed?

Reply:
Yes, it is. I'll publish the first beta version (= 0.99.0) today. So, the link will be alive within a few hours.